examples/distributed-spmv-scaling/distributed-spmv-scaling.cpp

```diff
@@ -0,0 +1,270 @@
+            std::cerr << "Usage: " << argv[0]
+                      << " [executor] [DISCRETIZATION_POINTS] [2D] "
+                         "[RESTRICT_STENCIL] [STRONG_SCALING]"
+                      << std::endl;
+            std::cerr << "Default values:" << std::endl;
+            std::cerr << "      - executor:        reference" << std::endl;
+            std::cerr << "      - DISCRETIZATION_POINTS: 100" << std::endl;
+            std::cerr << "      - 2D:                      1" << std::endl;
+            std::cerr << "      - RESTRICT_STENCIL:        0" << std::endl;
+            std::cerr << "      - STRONG_SCALING:          1" << std::endl;
+        }
+        std::exit(-1);
+    }
+
+    const auto executor_string = argc >= 2 ? argv[1] : "reference";
+    std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>>
+        exec_map{
+            {"omp", [] { return gko::OmpExecutor::create(); }},
+            {"cuda",
+             [local_rank] {
+                 return gko::CudaExecutor::create(
+                     local_rank, gko::ReferenceExecutor::create(), true);
+             }},
+            {"hip",
+             [local_rank] {
+                 return gko::HipExecutor::create(
+                     local_rank, gko::ReferenceExecutor::create(), true);
+             }},
+            {"dpcpp",
+             [local_rank] {
+                 return gko::DpcppExecutor::create(
+                     local_rank, gko::ReferenceExecutor::create());
+             }},
+            {"reference", [] { return gko::ReferenceExecutor::create(); }}};
+
+    // executor where Ginkgo will perform the computation
+    const auto exec = exec_map.at(executor_string)();  // throws if not valid
+
+    const auto dp = argc >= 3 ? atoi(argv[2]) : 100;
+    const bool two_dim = argc >= 4 ? atoi(argv[3]) > 0 : true;
+    const bool restricted = argc >= 5 ? atoi(argv[4]) > 0 : false;
+    const bool strong_scaling = argc >= 6 ? atoi(argv[5]) > 0 : true;
+
+    // Generate matrix data on each rank
+    if (rank == 0) {
+        std::cout << "Generating stencil matrix..." << std::endl;
+    }
+    auto A_data = two_dim ? generate_2d_stencil<ValueType, GlobalIndexType>(
+                                dp, comm, restricted, strong_scaling)
+                          : generate_3d_stencil<ValueType, GlobalIndexType>(
+                                dp, comm, restricted, strong_scaling);
+    const auto mat_size = A_data.size[0];
+    const auto rows_per_rank = mat_size / comm->size();
+
+    // build partition: uniform number of rows per rank
+    gko::Array<gko::int64> ranges_array{
+        exec->get_master(), static_cast<gko::size_type>(comm->size() + 1)};
+    for (int i = 0; i < comm->size(); i++) {
+        ranges_array.get_data()[i] = i * rows_per_rank;
+    }
+    ranges_array.get_data()[comm->size()] = mat_size;
+    auto partition = gko::share(
+        part_type::build_from_contiguous(exec->get_master(), ranges_array));
```

Comment:
We should probably make it explicit in the naming that this is a row partition. Either `gko::distributed::RowPartition` for the class name, or `part_type::build_from_contiguous_rows`, something like that.